sdk/core/Azure.Core/src/ResponseOfT.cs

```diff
@@ -6,7 +6,7 @@
 
 namespace Azure
 {
-    public readonly struct Response<T> : IDisposable
+    public readonly struct Response<T> : IDisposable where T : notnull
```

Comment:
>public readonly struct Response<T> : IDisposable where T : notnull [](start = 4, length = 66)

Should you actually guarantee that T is not null?  Is there a case where an empty body might not be an error condition?

Reply:
Good question! I'm tempted to say that having it `notnull` would work in most cases and save customers from thousands of null warnings. We can figure out corner cases on an individual basis.